cognite_toolkit/_cdf_tk/commands/modules.py

```
@@ -680,9 +709,62 @@ def add(self, organization_dir: Path) -> None:
             build_env = default.environment.validation_type
 
         existing_module_names = [module.name for module in ModuleResources(organization_dir, build_env).list()]
-        available_packages = Packages().load(self._builtin_modules_path)
-
+        available_packages = self._get_available_packages()
         added_packages = self._select_packages(available_packages, existing_module_names)
 
         download_data = self._get_download_data(added_packages)
         self._create(organization_dir, added_packages, environments, "update", download_data)
+
+    def _get_available_packages(self) -> Packages:
+        """
+        Returns a list of available packages, either from the CDF TOML file or from external libraries if the feature flag is enabled.
+        If the feature flag is not enabled and no libraries are specified, it returns the built-in modules.
+        """
+
+        if Flags.EXTERNAL_LIBRARIES.is_enabled() and CDFToml.libraries:
```

Should we raise an error here if we cannot find a library? Or do we have a default hardcoded one?

---

I thought about it and figured that if the users intends to use an external library, we shouldn't silently fall back to default. It is better to be explicit.